llarp/net/sock_addr.hpp

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <netinet/in.h>
+
+#include <string_view>
+#include <string>
+
+namespace llarp
+{
+  /// A simple SockAddr wrapper which provides a sockaddr_in (IPv4). Memory management is handled
+  /// in constructor and destructor (if needed) and copying is disabled.
+  struct SockAddr
+  {
+    SockAddr();
+    SockAddr(uint8_t a, uint8_t b, uint8_t c, uint8_t d);
+    SockAddr(uint8_t a, uint8_t b, uint8_t c, uint8_t d, uint16_t port);
+    SockAddr(std::string_view addr);
+
+    SockAddr(const SockAddr&);
+    SockAddr&
+    operator=(const SockAddr&) const;
+
+    SockAddr(const sockaddr& addr);
+    SockAddr(const sockaddr_in& addr);
+    operator const sockaddr*() const;
```

Comment:
I think this should be `operator sockaddr const*() const`, i.e. return a pointer to a const value.

Reply:
Err, `operator sockaddr* const() const`

Reply:
Wait, no, I am completely wrong about this.  Need more coffee.

Reply:
Heh, I struggled with that a bit yesterday. Boo pointers....